aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -221,7 +224,7 @@ private SyncTime filterOutOldSyncTimes(SyncTime lastSyncTime) throws DataStoreEx
                 BehaviorProcessor.createDefault(
                         appSync.buildSyncRequest(schema, lastSyncTimeAsLong, syncPageSize, predicate));
 
-        return processor.concatMap(request -> syncPage(request).toFlowable())
+        return processor.concatMap(request -> syncPageWithRetry(request).toFlowable())
```

Comment:
Have you thought of adding a feature flag to turn on/off retry?  It would be useful to quickly disable the feature or test this in subset of users.

Reply:
The android amplify implementation doesn't support feature flags at this time but we can make enable retry configurable by passing it as a property on the builder in the constructor.

Reply:
Added syncRetry enable as a property in builder which can be configured through constructor.